### DISCUSSION TITLE
feat: add a search page and tags page

### DISCUSSION THREAD

**Discussion by jerry011235** [hugo-sid/hugo-blog-awesome#51] [Feature requests]
It would be perfect to add a search page and tags page.

1. tags page refer to
[https://github.com/adityatelange/hugo-PaperMod/blob/master/layouts/_default/terms.html](https://github.com/adityatelange/hugo-PaperMod/blob/master/layouts/_default/terms.html)

2. search page demo (a very simple google site search)

2.1 Create `search.html` in `layouts\_default` folder.
```
{{ define "main"}}
<div class="wrapper">
    <form method="get" id="search-google" action="https://www.google.com/search" target="_blank">
       <input type="hidden" name="sitesearch" value="{{ .Site.BaseURL }}" />
       <input type="text" name="q" maxlength="255" value="" placeholder="Search" class="form-control" />
    </form>
</div>
{{ end }}
```
2.2 Create `search.md` in `blog\content` folder.
```
---
title: search
layout: "search"
---
```
2.3 config.toml
```
  [[menu.main]]
    pageRef="search"
    name = 'Search'
    url = '/search/'
    weight = 30	
```
Finally add favorite css and that's all.

**Comment by jerry011235** [comment]
Other small suggestions:
1. Post title(`.header-title`) and Post meta should be `text-align: center;`.

2. Add some additional information about the date format in `config.toml`, should tell people where to find the country date format.
```
# This defines how dates are formatted
# country_date_formats: https://gist.github.com/mlconnor/1887156
defaultContentLanguage = "en-gb"
```

**Comment by hugo-sid** [comment]
Thanks for the suggestions and detailed instructions @jerry011235.

Regarding tags, I am open to considering this. I will have a look at it when I get some time.
Regarding search, I am open to adding this feature but not the google site search way since:
- Google search isn't privacy friendly
- the search may not work if a site is not indexed by Google crawlers. We can't assume that every site is indexed by Google.

So, I prefer something which:
- doesn't require a visitor to take help of an external website (like Google, Duckduckgo etc.) for a simple search
- works for everyone (irrespective of their choice of getting the site indexed by Google)

This means we have to implement it using JS libraries like Fuse.js.

**Comment by jerry011235** [comment]
Hi, @hugo-sid,
> Google search isn't privacy friendly

I couldn't agree more, the privacy is really important.

If using the JS library,I suggest that Search Result Highlighting/Context, just like the effect below.This feels better for the user experience.
[https://koirand.github.io/blog/](https://koirand.github.io/blog/),search hugo for example.
[https://decovar.dev/blog/2020/01/05/hugo-search/](https://decovar.dev/blog/2020/01/05/hugo-search/)

I don't like the [PaperMod search](https://adityatelange.github.io/hugo-PaperMod/search/) effect , because there is no context on the search results page after entering a keyword(The keyword is in a post), and the keyword is not highlighted.

Other Information:
Google search alternative:
[https://github.com/mendel5/alternative-front-ends#google-search](https://github.com/mendel5/alternative-front-ends#google-search)
[https://github.com/pluja/awesome-privacy#search-engines](https://github.com/pluja/awesome-privacy#search-engines)

Third-Party Search Tools:
[https://manticoresearch.com/](https://manticoresearch.com/)
[https://www.algolia.com/](https://www.algolia.com/)

**Comment by Dieterbe** [comment]
i always liked how https://wordpress.com/theme/twentyfourteen does it. it shows a little search icon, click on it reveals an input field, when you submit it takes you to google results on your website.

**Comment by matrixik** [comment]
I would really like to have support for tags in posts.

**Comment by matrixik** [comment]
And regarding client site search there is for example this guide for Hugo (using https://www.fusejs.io/ ):
https://gist.github.com/cmod/5410eae147e4318164258742dd053993
